docs/internals/protocol/dataformats.rst

```diff
@@ -101,12 +101,23 @@ represented as:
 std::str
 ========
 
-The :eql:type:`std::str` values are represented as a UTF-8 encoded byte string.
+The :eql:type:`std::str` values are represented as a UTF-8 encoded byte string,
+prefixed with an ``int32`` length field:
+
+.. code-block:: c
+
+    struct String {
+        // Number of bytes.
+        int32       length;
```

Comment:
Why `int32` and not `uint32`?

Reply:
Actually, I don't think this is correct.  Length prefix is part of the wrapping structure, not the encoded data itself.

Reply:
https://github.com/edgedb/edgedb-js/blob/master/src/codecs/text.ts#L23-L31

Reply:
https://github.com/MagicStack/py-pgproto/blob/master/codecs/text.pyx#L24-L48